### ISSUE TITLE
Multiple stast field issue

### ISSUE THREAD

**Issue by vaseem-khan** [django-haystack/django-haystack#1210]
code from https://github.com/django-haystack/django-haystack/blob/master/haystack/backends/solr_backend.py#L247
 
```python
for k in stats.keys():
    kwargs['stats.field'] = k
    for facet in stats[k]:
                    kwargs['f.%s.stats.facet' % k] = facet
```
In case of multiple stats field at end of loop <code>kwargs['stats.field']</code> would only contain last stats key.

Shouldn't it would be like <code>kwargs['stats.field'] = stats.keys() </code> 
```python
kwargs['stats.field'] = stats.keys() 
for k in stats.keys():
    for facet in stats[k]:
                    kwargs['f.%s.stats.facet' % k] = facet
```

so Kwargs contain all stats fleld instead of one.

**Comment by chelis** [comment]
I don't think that is the solution as solr needs multiple stats.field params sent in the query, but I believe I am struggling with the same issue, Haystack only returns stats for one field (the last one specified if you chain stats)
